src/controllers/admin/event.ts

```diff
@@ -99,6 +97,15 @@ export const CreateEventHandler = async (req: Request, res: Response) => {
 }
 
 export const EditEventHandler =async (req : Request, res : Response) => {
+
+  const validevent = EventEditValidator.safeParse(req.body);
+  if(!validevent.success){
+    console.log(validevent.error);
+    return res.status(400).json({
+      message: "Invalid Event data provided!"
+    });
+  }
+
   const id = req.params.eventId;
```

Comment:
Query params has not been validated. Can be a potential surface for attack. Attach a zod schema in-line (no-need for separate type) with safeParse mechanism.

Rest looks good.

Reply:
Issue resolved.